### ISSUE TITLE
”new_xyz“ in "densify_and_clone" in "PGSR/scene/gaussian_model.py"

### ISSUE THREAD

**Issue by kimwao** [zju3dv/PGSR#20]
Hello! I am reading the `gaussian_model.py code` and have a question.
In the `densify_and_clone` function:
When the point clouds need to be cloned due to gradient limitations, why do new coordinates（`new_xyz`） of the cloned point cloud need to be moved in this way, especially using scaling as `stds` to establish a normal distribution for the `samples`?
```python
if selected_pts_mask.sum() > 0:
    new_xyz = self._xyz[selected_pts_mask]

    stds = self.get_scaling[selected_pts_mask]
    means =torch.zeros((stds.size(0), 3),device="cuda")
    samples = torch.normal(mean=means, std=stds)
    
    rots = build_rotation(self._rotation[selected_pts_mask])
    new_xyz = torch.bmm(rots, samples.unsqueeze(-1)).squeeze(-1) + self.get_xyz[selected_pts_mask]

**Comment by danpeng2** [comment]
Hello, kimwao.
Drawing on the operation of split. The original clone operation of 3DGS performs the clone directly at the original location, which may not be optimal for the new clone point. Therefore, under the distribution of Gaussian points, a new position is resampled as the new clone point position, resulting in a more even distribution of point positions.

**Comment by kimwao** [comment]
Thank you, danpeng. 
I still don't quite understand why `torch.bmm(rots, samples.unsqueeze(-1))` is used to represent the displacement of new points.  Multiplying a rotation matrix with a Gaussian distribution, and the Gaussian distribution comes from `scale`， Is this to constrain the range of movement? This design of transformation seems a bit peculiar~

**Comment by kimwao** [comment]
I have another question about the `densify_and_split` function. 
```python
if selected_pts_mask.sum() + n_init_points > self.max_all_points:
    limited_num = self.max_all_points - n_init_points
    # 布尔索引，不满足条件的padded_grad设置为0
    padded_grad[~selected_pts_mask] = 0
    ratio = limited_num / float(n_init_points)
    threshold = torch.quantile(padded_grad, (1.0-ratio))
    selected_pts_mask = torch.where(padded_grad > threshold, True, False)
    # print(f"split {selected_pts_mask.sum()}, raddi2D {padded_max_radii2D.max()} ,{padded_max_radii2D.median()}")
else:
    padded_grads_abs[selected_pts_mask] = 0
    mask = (torch.max(self.get_scaling, dim=1).values > self.percent_dense*scene_extent) & (padded_max_radii2D > self.abs_split_radii2D_threshold)
    padded_grads_abs[~mask] = 0
    selected_pts_mask_abs = torch.where(padded_grads_abs >= grad_abs_threshold, True, False)
    limited_num = min(self.max_all_points - n_init_points - selected_pts_mask.sum(), self.max_abs_split_points)
    if selected_pts_mask_abs.sum() > limited_num:
        ratio = limited_num / float(n_init_points)
        threshold = torch.quantile(padded_grads_abs, (1.0-ratio))
        selected_pts_mask_abs = torch.where(padded_grads_abs > threshold, True, False)
    selected_pts_mask = torch.logical_or(selected_pts_mask, selected_pts_mask_abs)
```
In the first `else` condition: 
 `padded_grads_abs[selected_pts_mask] = 0`  should be changed to 
`padded_grads_abs[~selected_pts_mask] = 0`.
My understanding is that in the `else` case, it need to add more points to the mask selection. The previously selected points are indicated by `selected_pts_mask` plus the new condition filter `mask`, so it should be `padded_grads_abs[~selected_pts_mask] = 0`. If my understanding is wrong, please correct me. Thank you!

**Comment by kimwao** [comment]
Oh, I see, it's written the same way in the 3dgs source code, where a Gaussian distribution defines a range of displacement. Thank U~
